src/clr-addons/slider/slider.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2018 Porsche Informatik. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild } from '@angular/core';
+import { Orientation } from './orientation.enum';
+import { SliderSize } from './slidersize.enum';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+
+@Component({
+  selector: 'clr-slider',
+  templateUrl: './slider.html',
+})
+export class ClrSlider {
+  // ======================
+  // ====== Children ======
+  // ======================
+  @ViewChild('slider') slider: ElementRef;
+  @Output('clrValueChanged') onValueChanged: EventEmitter<any> = new EventEmitter(false);
+
+  // ====================
+  // ====== Events ======
+  // ====================
+  valueChanged(value: number): void {
+    this.onValueChanged.emit(value);
+    this.value = value;
+  }
+
+  onMinValueChange(newMinValue: number): void {
+    this.minValue = newMinValue;
+  }
+
+  onMaxValueChange(newMaxValue: number): void {
+    this.maxValue = newMaxValue;
+  }
+
+  // ====================
+  // == Private Fields ==
+  // ====================
+  private _value = 0;
+  private _minValue = 0;
+  private _maxValue = 50;
+  private _step = 1;
+  private _showsLabels = true;
+  private _orientation: Orientation = Orientation.vertical;
+  private _size = SliderSize.medium;
+  private _enableValueFields = true;
+
+  // ====================
+  // ====== Getter ======
+  // ====================
```

Comment:
your getters seem unused - unused code should be removed.

generally you want to group each getter with a setter - instead of all setters & all getters